task/task.go

```diff
@@ -153,3 +186,30 @@ func flatten(message Message) Message {
 		return message
 	}
 }
+
+// router is an implementation of a `Sender` that is a resolver.
+type router struct {
+	resolverMu *sync.Mutex
+	resolver   Resolver
+}
+
+// NewResolver returns a new sender that represents a resolver. The given
+// resolver determines how the sender routes messages; any message `m` that is
+// sent to this sender will be sent to the sender determined by the resolver
+// through `Resolve(m)`.
+func NewResolver(resolver Resolver) Sender {
```

Comment:
Should this be called `NewRouter`? I feel like it makes more sense.

Reply:
Fixed in fa5cf78.